jujugui/static/gui/src/app/jujulib/test-payment.js

```diff
@@ -363,4 +370,221 @@ describe('jujulib payment service', function() {
+            line2: null,
+            state: 'Bunnyhug',
+            city: 'Sasquatch',
+            postcode: '90210',
+            country: 'North of the Border'
+          },
+          brand: 'Brand',
+          last4: '1234',
+          month: 3,
+          name: 'Main',
+          cardHolder: 'Mr G Spinach',
+          valid: true,
+          year: 2017
+        }]);
+        done();
+      });
+    });
+
+    it('handles errors when getting a user', (done) => {
+      const bakery = {
+        sendGetRequest: function(path, success, failure) {
+          const xhr = makeXHRRequest({Message: 'Uh oh!'});
+          failure(xhr);
+        }
+      };
+      const payment = new window.jujulib.payment('http://1.2.3.4/', bakery);
+      payment.getPaymentMethods('spinach', function(error, user) {
+        assert.equal(error, 'Uh oh!');
+        assert.isNull(user);
+        done();
+      });
+    });
+  });
+
+  describe('createPaymentMethod', () => {
+    it('can create a payment method', () => {
+      const originalMakeRequest = jujulib._makeRequest;
+      const makeRequest = sinon.stub();
+      jujulib._makeRequest = makeRequest;
+      const payment = new window.jujulib.payment('http://1.2.3.4/', {});
+      payment.createPaymentMethod('spinach', 'token123', sinon.stub());
+      // Restore the original method on the lib.
+      jujulib._makeRequest = originalMakeRequest;
+      assert.equal(makeRequest.callCount, 1);
+      assert.deepEqual(makeRequest.args[0][3], {
+        token: 'token123',
+        'payment-method-name': 'payment-method-created-2017-3-11-13-49-42-186'
+      });
+    });
+
+    it('can return the payment method when it has been created', (done) => {
+      const bakery = {
+        sendPutRequest: function(path, params, success, failure) {
+          assert.equal(
+            path,
+            'http://1.2.3.4/' +
+            window.jujulib.paymentAPIVersion +
+            '/u/spinach/payment-methods/' +
+            'payment-method-created-2017-3-11-13-49-42-186');
+          const xhr = makeXHRRequest({
+            address: {
+              name: 'Home',
+              line1: '1 Maple St',
+              line2: null,
+              county: 'Bunnyhug',
+              city: 'Sasquatch',
+              postcode: '90210',
+              country: 'North of the Border'
+            },
+            brand: 'Brand',
+            last4: '1234',
+            month: 3,
+            name: 'Main',
+            'card-holder': 'Mr G Spinach',
+            valid: true,
+            year: 2017
+          });
+          success(xhr);
+        }
+      };
+      const payment = new window.jujulib.payment('http://1.2.3.4/', bakery);
+      payment.createPaymentMethod('spinach', 'token123', (error, response) => {
+        assert.strictEqual(error, null);
+        assert.deepEqual(response, {
+          address: {
+            name: 'Home',
+            line1: '1 Maple St',
+            line2: null,
```

Comment:
Maybe mix it up a bit for the test and add some `line2`'s for some of them? Not sure if it'll catch anything/ make a difference.